src/System.Management.Automation/engine/parser/ast.cs

```diff
@@ -7400,10 +7382,7 @@ protected ExpressionAst(IScriptExtent extent)
         /// <returns></returns>
         internal virtual bool ShouldPreserveOutputInCaseOfException()
         {
-            var parenExpr = this as ParenExpressionAst;
-            var subExpr = this as SubExpressionAst;
-
-            if (parenExpr == null && subExpr == null)
+            if (this is not ParenExpressionAst or SubExpressionAst)
```

Comment:
Logic error

Reply:
@MartinGC94 Please address the comment.

Reply:
Thanks for the catch @xtqqczze 
Looking a bit more at that if statement, isn't it completely pointless? `NewInvalidOperationException` creates a new exception ready to be thrown but here it's neither thrown nor saved to a variable so it can be thrown later on.

Reply:
@MartinGC94 Please open new issue.

Reply:
Issue opened: #18812

Reply:
@MartinGC94 For future reference, logical patterns can use parentheses, e.g. `this is not (ParenExpressionAst or SubExpressionAst)`.